File: src/test/java/com/realyoungk/sdi/repository/GoogleSheetRepositoryTest.java
package com.realyoungk.sdi.repository;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;

import com.realyoungk.sdi.config.GoogleSheetsProperties;

@SpringBootTest
class GoogleSheetRepositoryTest {

    @Autowired
    private GoogleSheetRepository googleSheetRepository;
    @Autowired
    private GoogleSheetsProperties googleSheetsProperties;

    @Test
    void readSheetTitle() throws Exception {
        List<List<Object>> result = googleSheetRepository
                .readSheet(googleSheetsProperties.spreadsheetId(), googleSheetsProperties.titleRange());

        assertNotNull(result);
        assertEquals("날짜", result.getFirst().getFirst());
        assertEquals("기업명", result.getFirst().get(1));
        assertEquals("기수", result.getFirst().get(2));
        assertEquals("주선자", result.getFirst().get(3));
        assertEquals("참여 인원", result.getFirst().get(4));
        assertEquals("비고", result.getFirst().get(5));
    }
}

File: src/test/kotlin/com/realyoungk/sdi/repository/GoogleSheetRepositoryTest.kt
package com.realyoungk.sdi.repository

import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Disabled

class GoogleSheetRepositoryTest {

    @Test
    @Disabled("Google Sheets API 의존성으로 인해 비활성화 - 실제 환경에서만 테스트")
    fun `readSheet가 Google Sheets에서 데이터를 읽어온다`() {
        // 이 테스트는 실제 Google Sheets API 호출이 필요하므로
        // 자격 증명 파일과 네트워크 연결이 있는 환경에서만 활성화
        // 단위 테스트는 GoogleSheetRepository의 모킹된 버전을 사용
    }
}